File: stripe/src/test/java/com/stripe/android/exception/InvalidRequestExceptionTest.java
package com.stripe.android.exception;

import com.stripe.android.StripeErrorFixtures;

import org.junit.Test;

import java.net.HttpURLConnection;

import static org.junit.Assert.assertEquals;

public class InvalidRequestExceptionTest {

    @Test
    public void getStripeError_shouldReturnStripeError() {
        final StripeException stripeException = new InvalidRequestException(null, null, null,
                HttpURLConnection.HTTP_BAD_REQUEST, null, null,
                StripeErrorFixtures.INVALID_REQUEST_ERROR, null);
        assertEquals(StripeErrorFixtures.INVALID_REQUEST_ERROR, stripeException.getStripeError());
    }
}

File: stripe-core/src/test/java/com/stripe/android/core/exception/InvalidRequestExceptionTest.kt
package com.stripe.android.core.exception

import com.stripe.android.core.StripeErrorFixtures
import java.net.HttpURLConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class InvalidRequestExceptionTest {
    @Test
    fun stripeError_shouldReturnStripeError() {
        val stripeException = InvalidRequestException(
            StripeErrorFixtures.INVALID_REQUEST_ERROR,
            "req_123",
            HttpURLConnection.HTTP_BAD_REQUEST
        )
        assertEquals(StripeErrorFixtures.INVALID_REQUEST_ERROR, stripeException.stripeError)
    }

    @Test
    fun init_shouldDefaultMessageToStripeErrorMessage() {
        assertEquals(
            "This payment method (bancontact) is not activated for your account.",
            InvalidRequestException(StripeErrorFixtures.INVALID_REQUEST_ERROR).message
        )
    }

    @Test
    fun toString_withRequestId() {
        val actual = InvalidRequestException(
            requestId = "req_123",
            cause = IllegalArgumentException()
        ).toString()
        val expected =
            """
            Request-id: req_123
            com.stripe.android.core.exception.InvalidRequestException
            """.trimIndent()
        assertEquals(expected, actual)
    }

    @Test
    fun toString_withoutRequestId() {
        val actual = InvalidRequestException(
            cause = IllegalArgumentException()
        ).toString()
        val expected = "com.stripe.android.core.exception.InvalidRequestException"
        assertEquals(expected, actual)
    }
}
